chain-tx-enclave-next/mls/src/astree.rs

```diff
@@ -157,10 +158,8 @@ pub struct GroupKeySource<D: Input + BlockInput + FixedOutput + Reset + Default
 impl<D: Input + BlockInput + FixedOutput + Reset + Default + Clone> GroupKeySource<D> {
     pub fn new(application_secret: SecretVec<u8>, leaf_count: LeafSize) -> Self {
         let base_secret_source = TreeBaseKeySource::new(application_secret, leaf_count);
-        let mut ratchets = Vec::with_capacity(leaf_count.0);
-        for _ in 0..leaf_count.0 {
-            ratchets.push(None);
-        }
+        let mut ratchets = Vec::new();
```

Comment:
let mut ratchets: Vec<..>=vec![]  
as one line?

Reply:
it can't do that because the secret type don't implement `Clone`.